app/controllers/concerns/curation_concerns/application_controller_behavior.rb

```diff
@@ -7,20 +7,40 @@ module ApplicationControllerBehavior
     included do
       helper CurationConcerns::MainAppHelpers
 
-      rescue_from ActiveFedora::ObjectNotFoundError do |_exception|
-        render file: "#{Rails.root}/public/404", format: :html, status: :not_found, layout: false
+      rescue_from ActiveFedora::ObjectNotFoundError do |exception|
+        respond_to do |wants|
+          wants.html { render file: "#{Rails.root}/public/404", format: :html, status: :not_found, layout: false }
+          wants.json { render_json_response(response_type: :not_found, message: exception.message) }
+        end
       end
     end
 
     def deny_access(exception)
-      if [:show, :edit, :update, :destroy].include? exception.action
-        render 'unauthorized', status: :unauthorized
-      elsif current_user && current_user.persisted?
-        redirect_to main_app.root_url, alert: exception.message
+      if current_user && current_user.persisted?
+        respond_to do |wants|
+          wants.html do
+            if [:show, :edit, :update, :destroy].include? exception.action
+              render 'unauthorized', status: :unauthorized
+            else
+              redirect_to main_app.root_url, alert: exception.message
+            end
+          end
+          wants.json { render_json_response(response_type: :forbidden, message: exception.message) }
+        end
       else
         session['user_return_to'.freeze] = request.url
-        redirect_to main_app.new_user_session_path, alert: exception.message
+        respond_to do |wants|
+          wants.html { redirect_to main_app.new_user_session_path, alert: exception.message }
+          wants.json { render_json_response(response_type: :unauthorized, message: exception.message) }
+        end
       end
     end
+
+    # render a json response for +response_type+
+
+    def render_json_response(response_type: :success, message: nil, options: {})
+      json_body = CurationConcerns::API.generate_response_body(response_type: response_type, message: message, options: options)
+      render json: json_body, status: response_type
```

Comment:
Do we want to render the `location:` for creates and updates.  As done here:
https://github.com/rails/jbuilder/blob/master/lib/generators/rails/templates/controller.rb#L37